go/pkg/client/cas.go

```diff
@@ -640,7 +640,11 @@ func (c *Client) BatchWriteBlobs(ctx context.Context, blobs map[digest.Digest][]
 			return e
 		})
 		if err != nil {
-			return err
+			s, _ := status.FromError(err)
+			log.Errorf("FOOOOOO%v", status.Code(err))
+			log.Errorf("FOOOOOOO%v", s.Code())
+			log.Errorf("FOOOOO%v", err)
+			return err //status.Error(status.Code(err), fmt.Sprintf("error while uploading %d blobs, with total size of %d. Likely the client attempted to upload more than the server supported limit: %v", len(blobs), sz, err))
```

Comment:
(some offline discussion was involved)

why is this commented out?
IIRC that should solve your problem.

You can also do status.Errorf to avoid using fmt.Sprintf